system/Validation/CreditCardRules.php

```diff
@@ -245,9 +245,11 @@ public function valid_cc_number(string $ccNumber = null, string $type): bool
 		}
 
 		// Make sure it's a valid length for this card
-		$lengths = explode(',', $info['length']);
+		$lengths = array_map(function ($value) {
```

Comment:
Is this one going to be worth the extra effort to ensure strict types?

Reply:
phpstan complain about that, I may add `@phpstan-ignore-line` ignore it if you want.

Reply:
It seems casting to int is required:

```php
$info = [
    'length' => '1,2,3',  
];

$lengths = explode(',', $info['length']);
var_dump(in_array(1, $lengths, true));
var_dump(in_array('1', $lengths, true));

$lengths = array_map(function ($value) {
    return (int) $value;
}, explode(',', $info['length']));
var_dump(in_array(1, $lengths, true));
```

Got the following output:

```
bool(false)
bool(true)
bool(true)
```

see https://3v4l.org/lPubnY

Reply:
another solution is cast the `strlen()` itself to string.

```php
if (! in_array((string) strlen($ccNumber), $lengths, true))
```

Reply:
Updated to cast the `strlen()` value itself to string. It should be ok now.

Reply:
I meant if the strict version of `in_array` was worth the trouble of a custom array mapping, for performance and code clarity.